secant/Dependencies/SDKSynchronizer/SDKSynchronizerLive.swift

```diff
@@ -10,33 +10,42 @@ import Combine
 import ComposableArchitecture
 import ZcashLightClientKit
 
-enum SDKSynchronizerDependency: DependencyKey {
-    static let liveValue: SDKSynchronizerClient = LiveSDKSynchronizerClient()
-}
-
-class LiveSDKSynchronizerClient: SDKSynchronizerClient {
+private class SDKSynchronizerLiveWrapper {
     private var cancellables: [AnyCancellable] = []
-    private(set) var synchronizer: SDKSynchronizer?
-    // TODO: [#497] Since 0.17.0-beta SDKSynchronizer has `lastState` property which does exactly the same as `stateChanged`. Problem is that we have
-    // synchronizer as optional. And now it would be complicated to handle the situation when `lastState` isn't always available. Let's handle this
-    // in future.
-    private(set) var stateChanged: CurrentValueSubject<SDKSynchronizerState, Never>
-    private(set) var notificationCenter: NotificationCenterClient
-    private(set) var walletBirthday: BlockHeight?
-    private(set) var latestScannedSynchronizerState: SDKSynchronizer.SynchronizerState?
-
-    init(notificationCenter: NotificationCenterClient = .live) {
-        self.notificationCenter = notificationCenter
+    let synchronizer: SDKSynchronizer
+    let stateChanged: CurrentValueSubject<SDKSynchronizerState, Never>
+    var latestScannedSynchronizerState: SDKSynchronizer.SynchronizerState?
+
+    init(
+        notificationCenter: NotificationCenterClient = .live,
+        databaseFiles: DatabaseFilesClient = .liveValue,
```

Comment:
I am not sure if this way of using TCA dependencies is ok or not. It works though.